pkg/api/pod/util_test.go

```diff
@@ -338,3 +339,281 @@ func TestDropAlphaVolumeDevices(t *testing.T) {
 		}
 	}
 }
+
+func TestDropRunAsGroupField(t *testing.T) {
+	groupID := int64(2002)
+
+	testPod := api.Pod{
+		Spec: api.PodSpec{
+			SecurityContext: &api.PodSecurityContext{
+				RunAsGroup: &groupID,
+			},
+			Containers: []api.Container{
+				{
+					Name:  "container1",
+					Image: "testimage1",
+					SecurityContext: &api.SecurityContext{
+						RunAsGroup: &groupID,
+					},
+				},
+				{
+					Name:  "container2",
+					Image: "testimage2",
+					SecurityContext: &api.SecurityContext{
+						RunAsGroup: &groupID,
+					},
+				},
+			},
+			InitContainers: []api.Container{
+				{
+					Name:  "container1",
+					Image: "testimage1",
+					SecurityContext: &api.SecurityContext{
+						RunAsGroup: &groupID,
+					},
+				},
+			},
+		},
+	}
+
+	// Enable feature RunAsGroup
+	err1 := utilfeature.DefaultFeatureGate.Set("RunAsGroup=true")
+	if err1 != nil {
+		t.Fatalf("Failed to enable feature gate for RunAsGroup: %v", err1)
+	}
+
+	// now test dropping the fields - should not be dropped
+	DropDisabledAlphaFields(&testPod.Spec)
+
+	// check to make sure RunAsGroup is still present
```

Comment:
what if `RunAsGroup ` not exist? Have not seen any behavior.

Reply:
Done